SharedModules/mpi_func.f90

```diff
@@ -0,0 +1,198 @@
+module mpi_func
+  use numPrecision
+#ifdef MPI
+  use mpi_f08
+#endif
+  use errors_mod,        only : fatalError
+
+  implicit none
+
+  integer(shortInt), private   :: worldSize = 1
+  integer(shortInt), private   :: rank = 0
+  integer(shortInt), parameter :: MASTER_RANK = 0
+
+  !! Public type that replicates exactly particleState
+  !!
+  !! It is necessary for load balancing in the dungeon: particles have to be
+  !! transferred betwen processes, and MPI doesn't allow to transfer types with
+  !! type-bound procedures
+  type, public :: particleStateDummy
+    real(defReal)              :: wgt
+    real(defReal),dimension(3) :: r
+    real(defReal),dimension(3) :: dir
+    real(defReal)              :: E
+    integer(shortInt)          :: G
+    logical(defBool)           :: isMG
+    integer(shortInt)          :: type
+    real(defReal)              :: time
+    integer(shortInt)          :: matIdx
+    integer(shortInt)          :: cellIdx
+    integer(shortInt)          :: uniqueID
+    integer(shortInt)          :: collisionN
+    integer(shortInt)          :: broodID
+  end type particleStateDummy
+
+  !! Common MPI types
+#ifdef MPI
+  type(MPI_Datatype)   :: MPI_PARTICLE_STATE
+#endif
+
+contains
+
+  !!
+  !! Initialise MPI environment
+  !!
+  !! Needs to be called at the beginning of calculation before any MPI calls
+  !!
+  subroutine mpiInit()
+#ifdef MPI
+    integer(shortInt)        :: ierr
+
+    call mpi_init(ierr)
```

Comment:
As before, I think we need to be consistent with how we call MPI functions. Maybe I have missed something.

Reply:
We should. I am not sure if @Mikolaj-A-Kowalski had a plan when starting MPI. We could use full caps for the in-build MPI functions, like MPI_REDUCE and MPI_BCAST, and normal SCONE capitalisation for the SCONE procedures inside mpi_func.

Reply:
Yep... massive delay here. 
I am generally fully in the team `small_letters` with regards to all function calls. Capitals should just indicate constants etc. 
I would just argue that the ALL CAPS is a bit of a legacy thing from the 80s. We should just treat MPI procedures like we do the intrinsic procedures and keep them small.

Reply:
I think all the procedures should now be in small letters. The types however are all caps, as well as MPI inbuilt stuff like ``MPI_COMM_WORLD``. I could make it all small if we prefer, that's not a problem.